### PR TITLE
feat: supportedAccountTypes -> adapter.getSupportedAccountTypes()

### LINKED ISSUE #557: update to use new caip2/19 identifiers provided by chain adapters

### PR DESCRIPTION
**Pull request by gomesalexandre**
## Description

This removes the `supportedAccountTypes` constant that now has a better home following #611.

## Notice

- [x] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/CONTRIBUTING.md) guide?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

Not an utility per se, but brings some additional goodness to the closed #557

## Risk

Possibly some regressions to anything account-related.
Did some general regression testing as well and didn't find anything.

## Testing

N/A

### CODE REVIEW

File: src/context/AppProvider/AppContext.tsx [hunk 1]
```diff
@@ -153,7 +156,10 @@ export const AppProvider = ({ children }: { children: React.ReactNode }) => {
               const bitcoin = assetsById[CAIP19]
 
               if (!bitcoin) continue
-              for (const accountType of supportedAccountTypes.bitcoin) {
+              const supportedAccountTypes = (
+                adapter as UTXOBaseAdapter<UTXOChainTypes>
```
Comment (gomesalexandre) [inline src/context/AppProvider/AppContext.tsx:160]: If we don't cast it, this is going to give us some tsc errors about calling a possibly undefined, non-callable method. It's either that or using the `?.` method access notation and casting the result 🤷🏽

File: src/context/AppProvider/AppContext.tsx [hunk 2]
```diff
@@ -4,6 +4,10 @@ import {
   toRootDerivationPath,
   utxoAccountParams,
 } from '@shapeshiftoss/chain-adapters'
+import {
+  UTXOBaseAdapter,
+  UTXOChainTypes,
+} from '@shapeshiftoss/chain-adapters/dist/utxo/UTXOBaseAdapter'
```
Comment (0xdef1cafe) [inline src/context/AppProvider/AppContext.tsx:10]: diiiiiist

File: src/context/TransactionsProvider/TransactionsProvider.tsx [hunk 3]
```diff
@@ -86,11 +85,14 @@ export const TransactionsProvider = ({ children }: TransactionsProviderProps): J
           })
           .map(async chain => {
             const adapter = chainAdapterManager.byChain(chain)
+            // this looks funky, but we need a non zero length array to map over
+            // where we consume it - it either looks weird here or in the consumption
+            // so...  ¯\_(ツ)_/¯
```
Comment (0xdef1cafe) [inline src/context/TransactionsProvider/TransactionsProvider.tsx:90]: i understand you moved my comment over but it doesn't make complete sense in this context now (it was written at the implementation side over the consumption side)

**Review by 0xdef1cafe — CHANGES_REQUESTED** [review]

File: src/context/AppProvider/AppContext.tsx (hunk 2 above)
Comment (gomesalexandre) [reply src/context/AppProvider/AppContext.tsx:10]: I'll ask the dev that changed removed all the dist occurrences to do the same here

File: src/context/TransactionsProvider/TransactionsProvider.tsx (hunk 3 above)
Comment (gomesalexandre) [reply src/context/TransactionsProvider/TransactionsProvider.tsx]: Yeah probably looks odd now, kept the first line only

**Review by 0xdef1cafe — APPROVED** [review]

File: src/context/TransactionsProvider/TransactionsProvider.tsx [hunk 4]
```diff
@@ -93,18 +92,19 @@ export const TransactionsProvider = ({ children }: TransactionsProviderProps): J
           })
           .map(async chain => {
             const adapter = chainAdapterManager.byChain(chain)
+            // this looks funky, but we need a non zero length array to map over
+            const supportedAccountTypes = adapter.getSupportedAccountTypes?.() ?? [undefined]
```
Comment (theoboldfrazier) [inline src/context/TransactionsProvider/TransactionsProvider.tsx:96]: Why must it be a non-zero length array?

**Review by theoboldfrazier — APPROVED** [review]
Approved. There's some code smells, in here, but I understand it may not have been written the best to begin with, and it looks like they're known.

**Review by 0xdef1cafe — APPROVED** [review]
tested working, LGTM

**Pull request merged**